src/core/shape_inference/include/psroi_pooling_shape_inference.hpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <cmath>
+
+#include "compare.hpp"
+#include "dimension_util.hpp"
+#include "openvino/op/roi_pooling.hpp"
+#include "roi_pooling_shape_inference.hpp"
+
+namespace ov {
+namespace op {
+namespace psroi_pooling {
+namespace validate {
+template <class TROIPooling, class TShape>
+void feat_input_shape(const TROIPooling* op, const TShape feat_shape) {
```

Comment:
Is this `TROIPooling` template function currently used with other type than the `v1::PSROIPooling` or the template is used to enable reusage in the future?

Reply:
For future use with next versions. But if some base will be created could be changed  to be base class instead.